### PR TITLE
feat: implement veneer metrics part 1 - implement MetricsApiTracerAdapter (#2631)

### PR DESCRIPTION
**Pull request by kolea2**
port to 1x branch

(cherry picked from commit bf8a7e3e2ae2b7eaa6ad4170f2f3ceb38ddfa48a)

### CODE REVIEW

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml [hunk 1]
```diff
@@ -27,12 +27,34 @@ limitations under the License.
     and writers in Google Cloud Dataflow.
   </description>
 
+  <dependencyManagement>
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml:30]: @igorbernstein2 PTAL at the changes made here - they now match what is in the main branch

File: bigtable-client-core-parent/bigtable-client-core/pom.xml [hunk 2]
```diff
@@ -212,15 +213,15 @@ limitations under the License.
     <dependency>
       <groupId>io.dropwizard.metrics</groupId>
       <artifactId>metrics-core</artifactId>
-      <version>${dropwizard.metrics.version}</version>
+      <version>${hbase1-metrics.version}</version>
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/pom.xml:216]: Does bigtable-client-core still need dropwizard metrics?

File: bigtable-client-core-parent/bigtable-metrics-api/pom.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+      <version>3.1.2</version>
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-metrics-api/pom.xml:35]: should you use the variable here?

File: bigtable-client-core-parent/bigtable-metrics-api/pom.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      <groupId>com.google.api</groupId>
+      <artifactId>api-common</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>${slf4j.version}</version>
+    </dependency>
+  </dependencies>
+</project>
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-metrics-api/pom.xml:47]: newline

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml [hunk 5]
```diff
@@ -47,7 +47,7 @@
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <version>30.0-jre</version>
+      <version>30.1.1-jre</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:50]: I dont think this is necessary

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

File: bigtable-client-core-parent/bigtable-client-core/pom.xml (hunk 2 above)
Comment (kolea2) [reply bigtable-client-core-parent/bigtable-client-core/pom.xml:216]: still has imports: https://github.com/googleapis/java-bigtable-hbase/blob/bigtable-1.x/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/ResourceLimiterStats.java#L18

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml (hunk 5 above)
Comment (kolea2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:50]: here is the enforcer output when at 30.0-jre:
```
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for com.google.guava:guava:30.0-jre paths to dependency are:
+-com.google.cloud.bigtable:bigtable-hbase-1.x-tools:1.22.1-SNAPSHOT
  +-com.google.guava:guava:30.0-jre
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-tools:1.22.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x-shaded:1.22.1-SNAPSHOT
    +-io.grpc:grpc-census:1.38.0
      +-com.google.guava:guava:30.1-android
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-tools:1.22.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x-shaded:1.22.1-SNAPSHOT
    +-io.grpc:grpc-census:1.38.0
      +-io.grpc:grpc-api:1.38.0
        +-com.google.guava:guava:30.1-android
, 
Require upper bound dependencies error for com.google.errorprone:error_prone_annotations:2.3.4 paths to dependency are:
+-com.google.cloud.bigtable:bigtable-hbase-1.x-tools:1.22.1-SNAPSHOT
  +-com.google.guava:guava:30.0-jre
    +-com.google.errorprone:error_prone_annotations:2.3.4
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-tools:1.22.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x-shaded:1.22.1-SNAPSHOT
    +-io.grpc:grpc-census:1.38.0
      +-com.google.errorprone:error_prone_annotations:2.4.0
and
+-com.google.cloud.bigtable:bigtable-hbase-1.x-tools:1.22.1-SNAPSHOT
  +-com.google.cloud.bigtable:bigtable-hbase-1.x-shaded:1.22.1-SNAPSHOT
    +-io.grpc:grpc-census:1.38.0
      +-io.grpc:grpc-api:1.38.0
        +-com.google.errorprone:error_prone_annotations:2.4.0
```

Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:50]: in bigtable-hbase-1.x-tools/pom.xml, exclude grpc-census from bigtable-hbase-1.x-shaded. 

grpc-census gets shaded in bigtable-hbase-1.x-shaded and shouldnt leak here, but it does because of MNG-5899 & MSHADE-206

Comment (kolea2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml]: thank you! Fixed!

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-hadoop/pom.xml [hunk 6]
```diff
@@ -57,6 +57,10 @@ limitations under the License.
           <groupId>io.opencensus</groupId>
           <artifactId>*</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>io.grpc</groupId>
+          <artifactId>grpc-api</artifactId>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-hadoop/pom.xml:62]: you want to exclude grpc-census not api

File: pom.xml [hunk 7]
```diff
@@ -56,8 +56,9 @@ limitations under the License.
     <compileSource.1.8>1.8</compileSource.1.8>
 
     <!-- core dependency versions -->
-    <bigtable.version>1.19.2</bigtable.version>
-    <dropwizard.metrics.version>3.2.6</dropwizard.metrics.version>
+    <bigtable.version>1.27.1</bigtable.version>
```
Comment (igorbernstein2) [inline pom.xml:59]: can we move the version bump to a separate PR?

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

**Review by igorbernstein2 — APPROVED** [review]
lgtm
